File: code_snippets/ts/src/develop/workflows/service.ts
import * as restate from "@restatedev/restate-sdk";
import { SignUpWorkflow } from "./signup";
import { Context } from "@restatedev/restate-sdk";

export default restate.service({
  name: "user-management",
  handlers: {
    // <start_here>
    signUpUser: async (ctx: Context, email: string) => {
      // !focus(1:3)
      const result = await ctx
        .workflowClient<SignUpWorkflow>({ name: "signup" }, "someone")
        .run({ email });
    },
    queryStatus: async (ctx: Context) => {
      // !focus(1:3)
      const status = await ctx
        .workflowClient<SignUpWorkflow>({ name: "signup" }, "someone")
        .getStatus();
    },
    // <end_here>
  },
});

File: code_snippets/python/src/develop/workflows/service.py
from restate import VirtualObject, ObjectContext

from src.develop.workflows.signup import run, get_status

user_management_object = VirtualObject("UserManagementObject")


# <start_here>
@user_management_object.handler()
async def signup_user(ctx: ObjectContext, email: str):
    # !focus
    result = await ctx.workflow_call(run, key="someone", arg=email)


@user_management_object.handler()
async def query_status(ctx: ObjectContext):
    # !focus
    status: str = await ctx.workflow_call(get_status, key="someone", arg=None)


# <end_here>
